packages/flutter/lib/src/material/paginated_data_table.dart

```diff
@@ -106,6 +108,11 @@ class PaginatedDataTable extends StatefulWidget {
          return true;
        }()),
        assert(source != null),
+       assert(!(controller != null && (primary ?? false)),
+          'Primary ScrollViews obtain their ScrollController via inheritance from a PrimaryScrollController widget. '
+          'You cannot both set primary to true and pass an explicit controller.',
+       ),
+       primary = primary ?? controller == null,
```

Comment:
Why does it default to being primary? This could cause issues if another scrollview is using the PrimaryScrollController.

Reply:
fixed